sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/VpnSite.java

```diff
@@ -0,0 +1,170 @@
+
+    /** @return returns true if current VPN site is a security site. */
+    boolean isSecuritySite();
+
+    /** @return the AddressPrefixes value of VPN site. */
+    List<String> addressPrefixes();
+
+    /** @return the virtualWan property of VPN site. */
+    VirtualWan virtualWan();
+
+    /** @return the vpnSiteLinks property of VPN site. */
+    List<VpnSiteLinkInner> vpnSiteLinks();
+
+    /** @return the o365Policy property of VPN site. */
+    O365PolicyProperties o365Policy();
+
+    /** @return the device property of VPN site. */
+    DeviceProperties device();
+
+    /** The entirety of a VPN definition. */
+    interface Definition
+        extends DefinitionStages.Blank,
+        DefinitionStages.WithGroup,
+        DefinitionStages.WithCreate {
+    }
+
+    /** Grouping of VPN definition stages. */
+    interface DefinitionStages {
+        /**
+         * The first stage of a VPN definition.
+         */
+        interface Blank extends GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of the VPN definition allowing to specify the resource group.
+         */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<DefinitionStages.WithCreate> {
+        }
+
+        /**
+         * The stage of a route VPN definition which contains all the minimum required inputs for the resource to be
+         * created (via {@link VpnSite.DefinitionStages.WithCreate#create()}), but also allows for any other optional settings to be specified.
+         */
+        interface WithCreate extends Creatable<VpnSite>, DefinitionWithTags<WithCreate> {
+
+            /**
+             * Set the virtualWan property: The id of VirtualWAN to which the vpn belongs.
+             *
+             * @param subResourceId the id of virtualWan value to set.
+             * @return the next stage of the definition
+             */
+            WithCreate withVirtualWan(String subResourceId);
+
+            /**
+             * Set the addressSpace property: The AddressSpace that contains an array of IP address ranges.
+             *
+             * @param cidr the addressSpace value to set.
+             * @return the next stage of the definition
+             */
+            WithCreate withAddressSpace(String cidr);
+
+            /**
+             * Set the isSecuritySite property: IsSecuritySite flag.
+             *
+             * @param isSecuritySite the isSecuritySite value to set.
+             * @return the next stage of the definition
+             */
+            WithCreate withIsSecuritySite(Boolean isSecuritySite);
+
+            /**
+             * Set the vpnSiteLinks property: List of all vpn site links.
+             *
+             * @param vpnSiteLinks the vpnSiteLinks value to set.
+             * @return the next stage of the definition
+             */
+            WithCreate withVpnSiteLinks(List<VpnSiteLinkInner> vpnSiteLinks);
```

Comment:
@weidongxu-microsoft I think we probably need to model this `VpnSiteLink`. 
![image](https://github.com/user-attachments/assets/3289f13a-af50-4a69-bb3a-5aa670548328)

It should be a ChildResource of `VpnSite`. Similar pattern: [ApplicationGateway](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ApplicationGateway.java)(parent) vs [ApplicationGatewayFrontend](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ApplicationGatewayFrontend.java)(child)

Reply:
Agree.

Reply:
No, there isn't. It's created alongside `VpnSite`, though it can be get by an independent API.